internal/controller/standalone_pgadmin/controller.go

```diff
@@ -145,7 +148,7 @@ func (r *PGAdminReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		log.V(1).Info("Reconciled pgAdmin")
 	}
 
-	return ctrl.Result{}, err
+	return ctrl.Result{}, tracing.Escape(span, err)
```

Comment:
:thinking: Why is this using `Escape` instead of `Check`? Is it because we're leaving the current scope?

Reply:
> Is it because we're leaving the current scope?

Precisely.

You can add error information to a span separately from indicating the span failed. OTel has [this page](https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans) describing how to represent error information. Footnote **[3] `exception.escaped`** explains what "escape" means to them.

For us, yeah, escaping is likely that we're returning an error while ending a span.

`Check` adds error information without the "escaped" indicator.

Both functions do the thing we always have to do, `err != nil`, so we can use `Escape` kinda blindly. The code in this discussion/thread is doing `defer span.End()`, so all the early returns do an "add 'error escaped' when err != nil."

Reply:
:+1: Makes sense, thanks for the explanation.